client.go

```diff
@@ -19,11 +21,14 @@ func NewClient(xp Transporter, u ErrorUnwrapper) *Client {
 // the result field will be populated (if applicable). It returns an Error
 // on error, where the error might have been unwrapped from Msgpack via the
 // UnwrapErrorFunc in this client.
-func (c *Client) Call(method string, arg interface{}, res interface{}) (err error) {
+func (c *Client) Call(ctx context.Context, method string, arg interface{}, res interface{}) (err error) {
 	var d dispatcher
 	go c.xp.Run()
 	if d, err = c.xp.getDispatcher(); err == nil {
-		err = d.Call(method, arg, res, c.errorUnwrapper)
+		if ctx == nil {
```

Comment:
Generally speaking whenever we accept an interface it can be `nil` while satisfying the interface (something I really don't like about Go). Because of this, I'd prefer to check for `nil` at the junction of the exported interface, and supply the default when that's provided.